src/lang/modifyAst/addFillet.test.ts

```diff
@@ -439,6 +441,115 @@ const extrude001 = extrude(50, sketch001)
   })
 })
 
+const runModifyAstWithFilletAndTagTest = async (
+  code: string,
+  selectionSnippets: Array<string>,
+  radiusValue: number,
+  expectedCode: string
+) => {
+  // ast
+  const astOrError = parse(code)
+  if (err(astOrError)) {
+    return new Error('AST not found')
+  }
+  const ast = astOrError as Program
+
+  // selection
+  const segmentRanges: Array<[number, number]> = selectionSnippets.map(
+    (selectionSnippet) => [
+      code.indexOf(selectionSnippet),
+      code.indexOf(selectionSnippet) + selectionSnippet.length,
+    ]
+  )
+  const selection: Selections = {
+    codeBasedSelections: segmentRanges.map((segmentRange) => ({
+      range: segmentRange,
+      type: 'default',
+    })),
+    otherSelections: [],
+  }
+
+  // radius
+  const radius: KclCommandValue = {
+    valueAst: createLiteral(radiusValue) as Expr,
+    valueText: radiusValue.toString(),
+    valueCalculated: radiusValue.toString(),
+  }
+
+  // programMemory and artifactGraph
+  await kclManager.executeAst({ ast })
```

Comment:
This is what would be causing the leakiness between tests. I have a couple ideas of how to fix it but none of them are great:
1) We somehow figure out a way to label these tests in the same way we add `@electron` tests for our e2e tests, and then we run them separately serially, forcing them to run one by one.
2) We add an abstractions specifically for tests that does that setup i.e.
```ts
beforeAll(async () => {
  await initPromise
  // THESE TEST WILL FAIL without VITE_KC_DEV_TOKEN set in .env.development.local
  await new Promise((resolve) => {
    engineCommandManager.start({
      token: VITE_KC_DEV_TOKEN,
      ...
```
But then also handles the `executeAst` such that if there's another execution running it forces the late ones to wait and executes them 1 by 1. This might randomly make some tests wait a long time, so we'd have to bump timeout times for all tests that use this abstraction, but aside from that it might be a good authoring experience.
3) We do something similar to what is done in artifactGraph.test.ts where it does the executions for test before the rest of the tests starts, that way programMemory and the artifactGraph for the tests are ready to go before hand, where they can all be done serially, but instead of only for one file, we make it work for all tests that need to use this. Authoring is not great for this because you would need to specify all of the files you want executed separately from the bulk of the tests, and I'm sure if it's as flexible as we need it to be since there might be some tests that really need to execute twice within the one tests.


Ultimately this steams from the fact that we really can only do one connection to the websocket because it all has to go through the one instance of the engineConnection afaik.

Reply:
Oh I thought of 4) we port the artifactGraph to rust. Since this is the main reason we need to do a real execution (connects to the dev engine).

Up until this point we've been largely able to use the mock execution, but the reason we can't do that here is because without doing a full execution, rust will never use the `sendModelingCommandFromWasm` method and that's what we use to build up the info needed for the artifactGraph, I did think we could make a mock class that also had this method etc, but rust points specifically to this module and method, maybe there's a way but not one I found. Anyway if it were in rust, it probably be made to work with mock executions.

CC @adamchalmers